libafl/src/stages/sync.rs

```diff
@@ -125,15 +126,23 @@ where
         let to_sync = sync_from_disk_metadata.left_to_sync.clone();
         log::debug!("Number of files to sync: {:?}", to_sync.len());
         for path in to_sync {
-            let input = (self.load_callback)(fuzzer, state, &path)?;
+            let input = (self.load_callback)(fuzzer, state, &path);
+            let input_is_invalid = matches!(input, Err(Error::InvalidInput(_)));
             // Removing each path from the `left_to_sync` Vec before evaluating
             // prevents duplicate processing and ensures that each file is evaluated only once. This approach helps
-            // avoid potential infinite loops that may occur if a file is an objective.
-            state
-                .metadata_mut::<SyncFromDiskMetadata>()
-                .unwrap()
-                .left_to_sync
-                .retain(|p| p != &path);
+            // avoid potential infinite loops that may occur if a file is an objective or an invalid input.
+            if input.is_ok() || input_is_invalid {
```

Comment:
This should use a proper `match` arm

Reply:
```rust
match input {
    Ok(input) => input,
    Err(Error::InvalidInput(_)) => {
       log::debug(..);
       continue;
    }
    Err(e) => return e
}
```

Reply:
@domenukk the file needs to be removed from the metadata.

Reply:
Doesn't it always have to be removed?

Reply:
OK _technically_ we could keep it on error, but it's kinda pointless

Reply:
Ok let's remove it then